plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/WlmModeManager.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.wlm.WlmMode;
+import org.opensearch.wlm.WorkloadManagementSettings;
+
+/**
+ * Central manager for maintaining the current WLM mode with updates from cluster settings.
+ */
+public class WlmModeManager {
+
+    private volatile WlmMode wlmMode;
+
+    /**
+     * Constructor to initialize and register listener for WLM mode changes.
+     * @param settings OpenSearch settings
+     * @param clusterSettings Cluster settings to register update listener
+     */
+    public WlmModeManager(Settings settings, ClusterSettings clusterSettings) {
+        this.wlmMode = WorkloadManagementSettings.WLM_MODE_SETTING.get(settings);
+        clusterSettings.addSettingsUpdateConsumer(WorkloadManagementSettings.WLM_MODE_SETTING, this::setWlmMode);
+    }
+
+    /**
+     * Check if WLM mode is ENABLED or MONITOR_ONLY.
+     * Throws an IllegalStateException if WLM mode is DISABLED.
+     * @param operationDescription A short text describing the operation, e.g. "create workload group".
+     */
+    public void verifyWlmNotDisabled(String operationDescription) {
```

Comment:
I think better and more readable function name for this would be `ensureWlmIsEnabled`